scripts/scil_tractogram_pairwise_comparison.py

```diff
@@ -0,0 +1,115 @@
+                             assert_outputs_exist,
+                             assert_output_dirs_exist_and_empty,
+                             add_processes_arg,
+                             add_verbose_arg,
+                             is_header_compatible_multiple_files,
+                             load_tractogram_with_reference,
+                             validate_nbr_processes)
+from scilpy.tractograms.tractogram_operations import tractogram_pairwise_comparison
+
+
+def _build_arg_parser():
+
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_tractogram_1',
+                   help='Input tractogram 1.')
+    p.add_argument('in_tractogram_2',
+                   help='Input tractogram 2.')
+
+    p.add_argument('--out_dir', default='',
+                   help='Directory where all output files will be saved. '
+                        '\nIf not specified, outputs will be saved in the '
+                        'current directory.')
+    p.add_argument('--out_prefix', default='out',
+                   help='Prefix for output files. Useful for distinguishing '
+                        'between different runs.')
+
+    p.add_argument('--in_mask', metavar='IN_FILE',
+                   help='Optional input mask.')
+    p.add_argument('--skip_streamlines_distance', action='store_true',
+                   help='Skip computation of the spatial distance between '
+                        'streamlines.')
+    add_processes_arg(p)
+    add_reference_arg(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    assert_inputs_exist(parser, [args.in_tractogram_1,
+                                 args.in_tractogram_2])
+    to_verify = [args.in_tractogram_1, args.in_tractogram_2]
+    if args.in_mask:
+        to_verify.append(args.in_mask)
+    is_header_compatible_multiple_files(parser, to_verify,
+                                        verbose_all_compatible=True,
+                                        reference=args.reference)
+
+    if args.out_prefix and args.out_prefix[-1] == '_':
+        args.out_prefix = args.out_prefix[:-1]
+    out_corr_filename = os.path.join(args.out_dir,
+                                     '{}_correlation.nii.gz'.format(args.out_prefix))
+    out_acc_filename = os.path.join(args.out_dir,
+                                    '{}_acc.nii.gz'.format(args.out_prefix))
+    out_diff_filename = os.path.join(args.out_dir,
+                                     '{}_diff.nii.gz'.format(args.out_prefix))
+    out_merge_filename = os.path.join(args.out_dir,
+                                      '{}_heatmap.nii.gz'.format(args.out_prefix))
+    assert_output_dirs_exist_and_empty(parser, args, [], optional=args.out_dir)
```

Comment:
I chose '' as a default since this does not check the current directory if it is empty, it simply creates the files (most common use case). If I put './' and the user uses -f it would delete everything which is dangerous.

Using '' as default is the same as not using --out_dir (write in the current directory (even if not empty), like most scripts).